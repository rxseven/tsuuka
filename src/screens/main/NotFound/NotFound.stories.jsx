import React from 'react';
import { storiesOf } from '@storybook/react';

import NotFound from './index';

const title = 'Screens/Main/Not Found (404)';
const component = <NotFound />;

storiesOf(title, module).add('default', () => component);
