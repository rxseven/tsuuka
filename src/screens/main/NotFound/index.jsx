import React from 'react';
import { Link } from 'react-router-dom';

import { Body, Document, Head } from 'components/common/base/Page';
import Card from 'components/common/base/Card';
import Layout from 'components/common/composite/Layout';
import PATHS from 'constants/routes/paths';

function NotFound() {
  return (
    <Document>
      <Head>
        <title>Page not found · 通貨換算ツール</title>
      </Head>
      <Body>
        <Layout adaptive>
          <Card>
            <Card.Body>
              <Card.Title>404</Card.Title>
              <Card.Text>This is not webpage you are looking for.</Card.Text>
              <Link to={PATHS.root}>Go back to Home page</Link>
            </Card.Body>
          </Card>
        </Layout>
      </Body>
    </Document>
  );
}

export default NotFound;
