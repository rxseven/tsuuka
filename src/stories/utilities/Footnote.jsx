import React from 'react';
import styled from 'styled-components';

const Frame = styled.div`
  bottom: 2rem;
  font-size: 14px;
  position: fixed;
`;

function Footnote({ children }) {
  return <Frame>{children}</Frame>;
}

export default Footnote;
