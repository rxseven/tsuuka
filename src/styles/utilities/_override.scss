// Reset button's default styles
button:focus {
  box-shadow: none !important;
  outline: 0 !important;
}
