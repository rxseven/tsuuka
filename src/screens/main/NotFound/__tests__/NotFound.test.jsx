import { factory } from 'tests/utilities';
import NotFound from '../index';

// Arrange
const source = {};

// Setup
function setup(props) {
  return factory(NotFound, source, props);
}

// Test suites
describe('<NotFound />', () => {
  it('should render without crashing', () => {
    setup();
  });

  it('should render the correct content', () => {
    const expected = { content: '404' };
    const { component } = setup();

    expect(component).toHaveTextContent(expected.content);
  });
});
