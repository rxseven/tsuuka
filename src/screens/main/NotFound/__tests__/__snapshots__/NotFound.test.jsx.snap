// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> Snapshot tests should render correctly 1`] = `
<div
  className="row justify-content-sm-center"
>
  <div
    className="col-md-10 col-lg-8"
  >
    <div
      className="card text-text-left mb-3"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          404
        </h5>
        <p
          className="card-text"
        >
          This is not webpage you are looking for.
        </p>
        <a
          href="/"
          onClick={[Function]}
        >
          Go back to Home page
        </a>
      </div>
    </div>
  </div>
</div>
`;
